support-frontend/app/admin/settings/ContributionTypes.scala

```diff
@@ -0,0 +1,39 @@
+package admin.settings
+
+import com.gu.support.encoding.JsonHelpers
+import io.circe.{Decoder, Encoder}
+
+sealed trait ContributionType
+case object ONE_OFF extends ContributionType
+case object MONTHLY extends ContributionType
+case object ANNUAL extends ContributionType
+
+case object ContributionType {
+  implicit val contributionTypeEncoder: Encoder[ContributionType] = Encoder.encodeString.contramap[ContributionType](_.toString)
+
+  implicit val contributionTypeDecoder: Decoder[ContributionType] = JsonHelpers.decodeStringAndCollect {
```

Comment:
is there a circe macro that could avoid this boilerplate? or maybe using enumeratum? just curious!

Reply:
good suggestion - looks like enumeratum solves this https://github.com/lloydmeta/enumeratum#circe
([explanation of why it doesn't do this out of the box](https://stackoverflow.com/a/37015184))